tools/StaticAnalysis/BreakingChangeAnalyzer/TypeMetadataHelper.cs

```diff
@@ -0,0 +1,481 @@
+
+                    // If the types are the same, compare their properties
+                    if (oldPropertyType.Equals(newPropertyType, StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If we have not previously seen this type, run this
+                        // method on the type
+                        if (!_typeSet.Contains(oldPropertyType))
+                        {
+                            _typeSet.Add(oldPropertyType);
+
+                            CompareTypeMetadata(cmdlet, oldTypeMetadata, newTypeMetadata, issueLogger);
+                        }
+                    }
+                    else
+                    {
+                        // If the type of the property has been changed, log an issue
+                        issueLogger.LogBreakingChangeIssue(
+                            cmdlet: cmdlet,
+                            severity: 0,
+                            problemId: ProblemIds.BreakingChangeProblemId.ChangedPropertyType,
+                            description: string.Format(Properties.Resources.ChangedPropertyTypeDescription,
+                                oldProperty, oldType.Name, oldPropertyType, newPropertyType),
+                            remediation: string.Format(Properties.Resources.ChangedPropertyTypeRemediation,
+                                oldProperty, oldPropertyType));
+                    }
+                }
+                else
+                {
+                    // If the property has been removed, log an issue
+                    issueLogger.LogBreakingChangeIssue(
+                        cmdlet: cmdlet,
+                        severity: 0,
+                        problemId: ProblemIds.BreakingChangeProblemId.RemovedProperty,
+                        description: string.Format(Properties.Resources.RemovedPropertyDescription,
+                            oldProperty, oldType.Name),
+                        remediation: string.Format(Properties.Resources.RemovedPropertyRemediation,
+                            oldProperty, oldType.Name));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Checks if the type of the parameter is an array or a generic, and makes sure there are no breaking changes.
+        /// If the type is not an array or a generic, it proceeds with the normal type checking with CompareTypeMetadata.
+        /// </summary>
+        /// <param name="cmdlet">The cmdlet whose parameter metadata is being checked for breaking changes.</param>
+        /// <param name="parameter">The parameter whose type metadata is being checked for breaking changes.</param>
+        /// <param name="oldTypeMetadata">The type metadata from the old (serialized) assembly.</param>
+        /// <param name="newTypeMetadata">The type metadata from the new assembly.</param>
+        /// <param name="issueLogger">ReportLogger that will keep track of issues found.</param>
+        public void CheckParameterType(
```

Comment:
Same comment.  If we create separate methods for array checks versus geenric type checks, it will make the code easier to read.  Also, this lokks like a lot of duplication with the previus method.  If we had separate methods for the individual types of comparison, we might be able to reuse them in both places.